src/extensions/default/InlineColorEditor/main.js

```diff
@@ -59,7 +67,45 @@ define(function (require, exports, module) {
         } while (match && (pos.ch < start || pos.ch > end));
 
         if (!match) {
-            return null;
+            // Check if the cursorLine has a CSS rule of type color
+            var cssPropertyName, semiColonPos, colonPos, colorValue, cursorLineSubstring, firstCharacterPos;
+
+            // Get the css property name after removing spaces and ":" so that we can check for it in the file ColorProperties.json
+            cssPropertyName = cursorLine.split(':')[0].trim();
+
+            if (!cssPropertyName || !properties[cssPropertyName]) {
+                return null;
+            }
+
+            if (properties[cssPropertyName].type === "color") {
+                colonPos = cursorLine.indexOf(":");
+                semiColonPos = cursorLine.indexOf(";");
+                if (semiColonPos !== -1) {
+                    // edit the color value of an existing css rule
+                    cursorLineSubstring = cursorLine.substring(colonPos + 1, semiColonPos);
+                    colorValue = cursorLineSubstring.trim();
+                    if (colorRegEx.test(colorValue)) {
+                        firstCharacterPos = cursorLineSubstring.search(/\S/);
+                        pos.ch = colonPos + 1 + firstCharacterPos;
+                        endPos = {line: pos.line, ch: semiColonPos};
+                    } else {
+                        return null;
+                    }
+                } else {
+                    // edit the color value of a new css rule
+                    pos.ch = colonPos + 1;
```

Comment:
I think we'd be OK to just make sure we end up with one space after the text that we add the new color value to. If someone adds in 3 spaces before firing up the colorpicker, it would sort of be a benefit to them to format the CSS in a better way.

I just want to avoid removing a space if it's already added.